chain/rust/src/byron/serialization.rs

```diff
@@ -18,10 +18,13 @@ impl cbor_event::se::Serialize for AddrAttributes {
         &self,
         serializer: &'se mut Serializer<W>,
     ) -> cbor_event::Result<&'se mut Serializer<W>> {
+        // NOTE: This was manually modified *slightly*
```

Comment:
@SebastienGllmt just to confirm is this the desired behavior? The tests fail without it and it's what the code in the old byron/serialization.rs has, but is not what the direct byron.cddl does. I completely skimmed over it in the old code and thought I was going crazy when debugging and comparing vs old code.

Reply:
Could be a mistake. I tried to manually check to make sure everything matched since the old byron.cddl was a mess and I had to fix multiple things in it, but maybe I missed this

Reply:
I assume it was on purpose. The code would have been manually edited after codegen before to be like that. It doesn't start in `Ae2` like byron addresses used to IIRC, but weren't there multiple kinds? (Icarus or not). I don't know as much about Byron as you do though so I'm not sure. It's possible that the change should be, instead of being in serialization, in the construction to avoid populating the stake distribution if it's bootstrap instead of including it in the first place. The one created by including the bootstrap stake distribution is `2657WMsDfac5izkoz9CMcGTDcLwq5KGcNcj5GT2HMxvy11CCNjj2ae7PvEAcqVivB`. Does that look like any kind of byron address? (non-icarus or whatever)

Reply:
FYI this is what is done in the old Rust implementation of Byron by IOG https://github.com/input-output-hk/rust-byron-cardano/blob/5f0c8d34424e0ccb199b8c6b70fd410d4bd6af41/cardano/src/address.rs#L257

The spec also has some mistakes in it, and you can see me talking about them here https://github.com/input-output-hk/cardano-ledger/pull/2923

Reply:
`2657WMsDfac5izkoz9CMcGTDcLwq5KGcNcj5GT2HMxvy11CCNjj2ae7PvEAcqVivB` Daedalus addresses usually start with Ae2 or Dd, but there are two cases where this isn't the case

1. On testnet, they have different prefixes (I think one of them started with 2?). Note Cardano didn't have testnet support at first, then it was added later (it was made backwards compatible by defining mainnet as having no protocol magic in the address)
2. Some Byron addresses have junk data stuffed into them (since note the address contains a map you can just put anything you want into them theoretically). This was then disallowed in a later version of Cardano

Reply:
It looks like that Rust implementation of Byron does the same thing (omits it entirely if bootstrap). This wouldn't be 2) since the only extra information was the bootstrap stake distribution. I don't think it's 1) either since the address is for mainnet.

Reply:
@rooooooooob let's add some testcases for these addresses starting with 2 and also for the long addresses

Reply:
@gostkin @SebastienGllmt Do we need test cases for these before we can merge this PR? I tried trawling through old issues/PRs and I found workarounds in input-output-hk/cardano-byron-cli#73 and Emurgo/project-icarus-importer#3 but no addresses are explicitly mentioned (also searching the block mentioned there seems to give normal address lengths - maybe the explorer trims them?)

I could sync through byron again and explicitly check but I don't think it's worth it. The addresses are _**huge**_ and there's no difference encoding-wise. There's just a plain `bytes` field inside the byron address' HD payload that has zero restrictions and can contain anything. I don't see what the test would accomplish besides maybe lagging your editor when you open `address.rs`. We've successfully round-tripped every byron block so I assume there's no issues.

I could add in a testnet one though, that should be easier to find.

Reply:
Actually it looks like we already have tests in `chain/rust/src/byron/utils.rs` to cover testnet addresses.